### PR TITLE
Update OpenAPI Spec and Fix Bug

### PR DESCRIPTION
**Pull request by karanataryn**

### CODE REVIEW

**Comment by MarkLindblad** [comment]
It looks like the `selected_pages` option does not support ranges, and only supports individual pages. It also looks like mintlify's web UI to demo the API might have a bug which makes the file part not be set correctly.

For example:
Screenshot 2024-11-12 at 10 09 08 AM
results in
```bash
curl --request POST \
  --url https://api.aryn.cloud/v1/document/partition \
  --header 'Authorization: Bearer REDACTED'\
  --header 'Content-Type: multipart/form-data' \
  --form 'file={}'
```
Which returns
`{"detail":[{"type":"value_error","loc":["body","file"],"msg":"Value error, Expected UploadFile, received: <class 'str'>","input":"[object Object]","ctx":{"error":{}}}]}`

The commands should instead look something like this:
```bash
curl --request POST \
  --url https://api.aryn.cloud/v1/document/partition \
  --header 'Authorization: Bearer REDACTED'\
  --header 'Content-Type: multipart/form-data' \
  --form 'file=@mergeme.pdf'
```

**Comment by MarkLindblad** [comment]
Try:
```
oneOf:
  - type: integer
  - type: array
    items:
      type: integer
    minItems: 2
    maxItems: 2
description: |
  A single integer (e.g., 1) or an array with exactly two integers representing a range (e.g., [1, 10]).
```

File: api-reference/openapi.yaml [hunk 1]
```diff
@@ -44,19 +44,43 @@ components:
...
-            - type: 'null'
-          title: Options
+          type: object
+          # additionalProperties: true
+          properties:
+            selected_pages:
+              type: array
+              items:
+                type: integer
+              title: Selected Pages
```
Comment (MarkLindblad) [inline api-reference/openapi.yaml:59]: This should instead be
```yaml
selected_pages:
  oneOf:
    - type: integer
    - type: array
      items:
        type: integer
      minItems: 2
      maxItems: 2
  description: |
    A single integer (e.g., 1) or an array with exactly two integers representing a range (e.g., [1, 10]).
```

Comment (karanataryn) [reply api-reference/openapi.yaml]: Done.

File: api-reference/openapi.yaml [hunk 2]
```diff
@@ -44,42 +44,94 @@ components:
...
+            selected_pages:
+              type: array
+              items:
+                oneOf:
+                  - type: integer
+                  - type: array
+                    items:
+                      type: integer
+              minItems: 2
+              maxItems: 2
```
Comment (MarkLindblad) [inline api-reference/openapi.yaml:64]: This is not indented properly. The min/max should apply to the innermost array.

Comment (karanataryn) [reply api-reference/openapi.yaml]: Ah good catch, I was doing some experimentation. Fixed.

File: api-reference/openapi.yaml [hunk 3]
```diff
@@ -44,42 +44,94 @@ components:
...
+                oneOf:
+                  - type: integer
+                  - type: array
+                    items:
+                      type: integer
+                    minItems: 2
+                    maxItems: 2
+              title: Selected Pages
+              description: |
+                An array containing single integers (e.g., 1) or arrays with exactly two integers representing a range (e.g., [1, 10]).
```
[the remark this answers is not in the document]
Comment (karanataryn) [reply api-reference/openapi.yaml:67]: Would rather keep this in the documentation to reduce clutter in the spec.

**Review by MarkLindblad — APPROVED** [review]

**Pull request merged**